protocol/contracts/beanstalk/metadata/MetadataImage.sol

```diff
@@ -524,13 +526,93 @@ contract MetadataImage {
-                uint256(stem).toString(),
+                '<text x="235" y="14.5" font-size="12" fill="White" text-anchor="end" font-family="futura">Stem: ',
+                SciNotation(stem),
                 '</text>'
             )
         );
     }
 
+    function SciNotation(int96 stem) internal pure returns (string memory) {
+        if(stem >= 0){
+            // if stem is greater than 1e7, use scientific notation
+            if(stem > 100_000){
+                return powerOfTen(uint256(stem));
+            } else {
+                return uint256(stem).toString();
+            }
+        } else {
+            // if stem is greater than 1e7, use scientific notation
+            if(-stem > 100_000){
+                return string(abi.encodePacked("-", powerOfTen(uint256(-stem))));
+            } else {
+                return int256(stem).toString();
+            }
+        }
+    }
+
+    function powerOfTen(uint256 stem) internal pure returns (string memory) {
+        // if else ladder to determine how many digits to show
+        if(stem < 1e6){
+            return stemDecimals(stem, 5);
+        } else if(stem < 1e7) {
+            return stemDecimals(stem, 6);
+        } else if(stem < 1e8) {
+            return stemDecimals(stem, 7);
+        } else if(stem < 1e9) {
+            return stemDecimals(stem, 8);
+        } else if(stem < 1e10) {
+            return stemDecimals(stem, 9);
+        } else if(stem < 1e11) {
+            return stemDecimals(stem, 10);
+        } else if(stem < 1e12) {
+            return stemDecimals(stem, 11);
+        } else if(stem < 1e13) {
+            return stemDecimals(stem, 12);
+        } else if(stem < 1e14) {
+            return stemDecimals(stem, 13);
+        } else if(stem < 1e15) {
+            return stemDecimals(stem, 14);
+        } else if(stem < 1e16) {
+            return stemDecimals(stem, 15);
+        } else if(stem < 1e17) {
+            return stemDecimals(stem, 16);
+        } else if(stem < 1e18) {
+            return stemDecimals(stem, 17);
+        } else if(stem < 1e19) {
+            return stemDecimals(stem, 18);
+        } else if(stem < 1e20) {
+            return stemDecimals(stem, 19);
+        } else if(stem < 1e21) {
+            return stemDecimals(stem, 20);
+        } else if(stem < 1e22) {
+            return stemDecimals(stem, 21);
+        } else if(stem < 1e23) {
+            return stemDecimals(stem, 22);
+        } else if(stem < 1e24) {
+            return stemDecimals(stem, 23);
+        } else if(stem < 1e25) {
+            return stemDecimals(stem, 24);
+        } else if(stem < 1e26) {
+            return stemDecimals(stem, 25);
+        } else if(stem < 1e27) {
+            return stemDecimals(stem, 26);
+        } else if(stem < 1e28) {
+            return stemDecimals(stem, 27);
+        } else {
+            return stemDecimals(stem, 28);
+        }
+    }
```

Comment:
can you a blank line between these functions?

Reply:
fixed: c58be9b7fa3eda06805602e40e1e93f0fb281137